font/src/ft/mod.rs

```diff
@@ -306,32 +285,49 @@ impl FreeTypeRasterizer {
         Ok(font_key)
     }
 
-    fn full_metrics(&self, key: FontKey) -> Result<FullMetrics, Error> {
-        let face = self.faces.get(&key).ok_or(Error::FontNotLoaded)?;
+    fn full_metrics(&self, face_load_props: &FaceLoadingProperties) -> Result<FullMetrics, Error> {
+        let ft_face = &face_load_props.ft_face;
+        let size_metrics = ft_face.size_metrics().ok_or(Error::MissingSizeMetrics)?;
 
-        let size_metrics = face.ft_face.size_metrics().ok_or(Error::MissingSizeMetrics)?;
-
-        let width = match face.ft_face.load_char('0' as usize, face.load_flags) {
-            Ok(_) => face.ft_face.glyph().metrics().horiAdvance / 64,
+        let width = match ft_face.load_char('0' as usize, face_load_props.load_flags) {
+            Ok(_) => ft_face.glyph().metrics().horiAdvance / 64,
             Err(_) => size_metrics.max_advance / 64,
         } as f64;
 
         Ok(FullMetrics { size_metrics, cell_width: width })
     }
 
+    fn load_ft_face(&mut self, path: PathBuf, index: isize) -> Result<Rc<FTFace>, Error> {
```

Comment:
We can inline this function now, what do you think about it?

Reply:
I don't see anything wrong with keeping it separate.